src/components/addCard/index.js

```diff
@@ -170,7 +170,7 @@ export default class AddCard extends Component {
             onChangeText={rawCardNumber => {
               const cardNumber = s(rawCardNumber).replaceAll(' ', '').s
               this.setState({ cardNumber: cardNumber })
-              if (payment.fns.validateCardNumber(cardNumber)) {
+              if (payment.fns.validateCardNumber(cardNumber) && cardNumber.length === 16) {
```

Comment:
Maybe do the condition this way round `(cardNumber.length === 16 && payment.fns.validateCardNumber(cardNumber) )`, that way the `validateCardNumber` function will only be called when the card is 16 digits long, which might give a bit of a performance increase